src/nix/nix_store.rs

```diff
@@ -127,11 +140,59 @@ impl NixStoreCmd {
                 .collect();
             Ok(out)
         } else {
-            let exit_code = out.status.code().unwrap_or(1);
-            bail!(
-                "nix-store --query --requisites --include-outputs failed to run (exited: {})",
-                exit_code
-            );
+            let stderr = String::from_utf8(out.stderr).ok();
+            let exit_code = out.status.code();
+            Err(CommandError::ProcessFailed { stderr, exit_code }.into())
         }
     }
 }
+
+/// Errors when running and interpreting the output of a nix command
+#[derive(Error, Debug)]
+pub enum NixCmdError {
```

Comment:
You shouldn't use duplicate names for types in order to avoid confusion. `NixCmdError` already exists: https://github.com/juspay/nix-rs/blob/2dbba678a9e67925d964dbde24fbd8951ecbd2de/src/command.rs#L187

```suggestion
pub enum NixStoreCmdError {
```